### PR TITLE
Phoenix backend

### PR DESCRIPTION
**Pull request by ivan-unfolds**
* Change node backend to phoenix backend

### CODE REVIEW

File: assets/js/socket.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// NOTE: The contents of this file will only be executed if
```
Comment (andrewMacmurray) [inline assets/js/socket.js]: should take this file out as we're unlikely to need sockets, plus if we're using elm we'd use the elm-phoenix channel library

File: assets/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "repository": {},
+  "license": "MIT",
+  "scripts": {
+    "deploy": "brunch build --production",
+    "watch": "brunch watch --stdin"
+  },
+  "dependencies": {
+    "phoenix": "file:../deps/phoenix",
```
Comment (andrewMacmurray) [inline assets/package.json]: elm as a dependency?

File: lib/tech_for_good_near_you/web/channels/user_socket.ex [hunk 3]
```diff
@@ -0,0 +1,37 @@
+defmodule TechForGoodNearYou.Web.UserSocket do
```
Comment (andrewMacmurray) [inline lib/tech_for_good_near_you/web/channels/user_socket.ex]: maybe remove this file too as we're unlikely to need websockets

File: lib/tech_for_good_near_you/web/endpoint.ex [hunk 4]
```diff
@@ -0,0 +1,57 @@
+defmodule TechForGoodNearYou.Web.Endpoint do
+  use Phoenix.Endpoint, otp_app: :tech_for_good_near_you
+
+  socket "/socket", TechForGoodNearYou.Web.UserSocket
```
Comment (andrewMacmurray) [inline lib/tech_for_good_near_you/web/endpoint.ex]: and remove this line for same socket reason

File: lib/tech_for_good_near_you/web/router.ex [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  pipeline :api do
+    plug :accepts, ["json"]
+  end
+
+  scope "/", TechForGoodNearYou.Web do
+    pipe_through :browser # Use the default browser stack
+
+    get "/", ElmController, :index
+    get "/events", EventController, :index
```
Comment (andrewMacmurray) [inline lib/tech_for_good_near_you/web/router.ex]: isn't this an api endpoint?

Comment (andrewMacmurray) [reply lib/tech_for_good_near_you/web/router.ex]: or is it for the admin view?

File: mix.exs [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  defp deps do
+    [{:phoenix, "~> 1.3.0-rc"},
+     {:phoenix_pubsub, "~> 1.0"},
+     {:phoenix_ecto, "~> 3.2"},
+     {:postgrex, ">= 0.0.0"},
+     {:phoenix_html, "~> 2.6"},
+     {:phoenix_live_reload, "~> 1.0", only: :dev},
+     {:gettext, "~> 0.11"},
+     {:cowboy, "~> 1.0"},
+     {:httpotion, "~> 3.0.2"}]
```
Comment (andrewMacmurray) [inline mix.exs]: there seems there's a consensus in the community on `HTTPoison` vs `HTTPotion`, I think we should use `HTTPoison` instead

https://elixirforum.com/t/httpoison-vs-httpotion/2221/10

Comment (ivan-unfolds) [reply mix.exs]: Can you make requests with `HTTPosion`? That's what I am using `HTTPotion` for in `event_controller`.

File: lib/tech_for_good_near_you/web/router.ex (hunk 5 above)
Comment (ivan-unfolds) [reply lib/tech_for_good_near_you/web/router.ex]: Good shout, will change.

File: mix.exs (hunk 6 above)
Comment (andrewMacmurray) [reply mix.exs]: They're effectively the same thing, but Poision seems to have better https support, also given that Phoenix is so convention driven it's probably best to stick to the consensus unless we really need Potion for some reason

File: assets/package.json (hunk 2 above)
Comment (ivan-unfolds) [reply assets/package.json]: Is this so that we can deploy?

File: mix.exs (hunk 6 above)
Comment (andrewMacmurray) [reply mix.exs]: The intercom client is a "thin wrapper over `HTTPoision`"

File: lib/tech_for_good_near_you/web/channels/user_socket.ex (hunk 3 above)
Comment (ivan-unfolds) [reply lib/tech_for_good_near_you/web/channels/user_socket.ex]: Sounds good, will get rid of sockets.

File: assets/package.json (hunk 2 above)
Comment (andrewMacmurray) [reply assets/package.json]: Yea we'll need that for building it on heroku, and also to make it explicit that we're using it. also if we want CI tests we'd need it as a dependency

File: mix.exs (hunk 6 above)
Comment (ivan-unfolds) [reply mix.exs]: That sounds good, I was looking at `Poison` rather than `HTTPoison` and getting quite confused 😊

Comment (ivan-unfolds) [reply mix.exs]: Will change it over.

File: assets/package.json (hunk 2 above)
Comment (ivan-unfolds) [reply assets/package.json]: That makes sense.

File: mix.exs (hunk 6 above)
Comment (andrewMacmurray) [reply mix.exs]: Ahh fair enough, similar names lol

**Comment by ivan-unfolds** [comment]
@andrewMacmurray I've made all the changes now.

**Review by andrewMacmurray — APPROVED** [review]
looks good

**Pull request merged**